src/http/web_socket/protocol.cr

```diff
@@ -229,7 +229,20 @@ class HTTP::WebSocket::Protocol
     end
   end
 
-  def close(message = nil)
+  def close(reason : String? = nil, code : Int16? = nil)
+    if code
+      raw = uninitialized UInt8[2]
+      IO::ByteFormat::BigEndian.encode(code, raw.to_slice)
+
+      message = String.new(raw)
```

Comment:
I can't benchmark this right now, but I don't think creating this string is necessary over just creating the slice we need to send and writing everything to that:

```cr
  slice = Bytes.new(2 + reason.bytesize)
  IO::ByteFormat::NetworkEndian.encode(code, slice)
  reason.to_slice.copy_to(slice + 2)

  # ..

  send(slice, Opcode::CLOSE)
```
https://carc.in/#/r/6dg3